src/lib_json/json_tool.h

```diff
@@ -101,6 +101,9 @@ template <typename Iter> Iter fixNumericLocale(Iter begin, Iter end) {
 }
 
 template <typename Iter> void fixNumericLocaleInput(Iter begin, Iter end) {
+#ifdef JSONCPP_NO_LOCALE_SUPPORT
+  return;
+#else
   char decimalPoint = getDecimalPoint();
   if (decimalPoint == '\0' || decimalPoint == '.') {
```

Comment:
In method `getDecimalPoint`, if JSONCPP_NO_LOCALE_SUPPORT the method will return `\0`, and then it will return directly here.

Reply:
The commit message goes over this. if JSONCPP_NO_LOCALE_SUPPORT, decimalPoint == '\0' will always be true, making it a pointless comparison.